crates/rolldown_binding/src/types/binding_output_chunk.rs

```diff
@@ -3,113 +3,255 @@ use std::collections::HashMap;
 use napi_derive::napi;
 use rolldown_sourcemap::SourceMap;
 
-use crate::types::binding_rendered_module::BindingRenderedModule;
+use crate::{type_aliases::WeakRefMutex, types::binding_rendered_module::BindingRenderedModule};
 
 #[napi]
 pub struct BindingOutputChunk {
-  inner: &'static mut rolldown_common::OutputChunk,
+  inner: WeakRefMutex<Vec<rolldown_common::Output>>,
+  index: usize,
 }
 
 #[napi]
 impl BindingOutputChunk {
-  pub fn new(inner: &'static mut rolldown_common::OutputChunk) -> Self {
-    Self { inner }
+  pub fn new(inner: WeakRefMutex<Vec<rolldown_common::Output>>, index: usize) -> Self {
+    Self { inner, index }
   }
 
   #[napi(getter)]
   pub fn is_entry(&self) -> bool {
-    self.inner.is_entry
+    self.inner.with_inner(|inner| {
```

Comment:
This looks ugly. But I can't think any good way on this except macros.

Reply:
We might able to transform `Vec<Output>` to `Vec<UniqueArcMutex<Asset>>` and `Vec<UniqueArcMutex<Chunk>>` to modelize the old struct. Need more investigation.